src/engine/qcommon/common.cpp

```diff
@@ -770,62 +771,37 @@ EVENT LOOP
 static const int MAX_QUEUED_EVENTS  = 1024;
 static const int MASK_QUEUED_EVENTS = ( MAX_QUEUED_EVENTS - 1 );
 
-static sysEvent_t eventQueue[ MAX_QUEUED_EVENTS ];
+static std::unique_ptr<Sys::EventBase> eventQueue[ MAX_QUEUED_EVENTS ];
```

Comment:
How come you didn't use a vector or list?

Reply:
I didn't really think about it, but it doesn't necessarily seem like a good thing for an unlimited amount of events to accumulate if the game is frozen or whatever.

Reply:
Hmm, it may be irrational but I am loath to replace a perfectly good array with some wacked out vector of linked list nodes thing which is std::queue.

Reply:
std::queue should be implemented with std::dequeue which is a bit smarter than that.

Reply:
What do you mean about deque being smarter than that? That's what I was trying to describe. Just the part where the linked list nodes are small arrays?

deque seems like such a pointless data structure to have... I guess it's some kind of workaround for the lack of unique_ptr in older versions? Otherwise you could just have a double-ended vector and put unique_ptrs in it.